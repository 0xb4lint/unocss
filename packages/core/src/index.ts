import { createGenerator as _createGenerator } from './generator'
import { resolveConfig as _resolveConfig } from './config'
import { UserConfigDefaults } from './types'

export * from './types'
export * from './utils'
export * from './extractors'

export type ArgumentType<T> = T extends ((...args: infer A) => any) ? A : never
export type Shift<T> = T extends [_: any, ...args: infer A] ? A : never
export type RestArgs<T> = Shift<ArgumentType<T>>

export function createMiniwindWithDefaults(defaults: UserConfigDefaults) {
  return {
    createGenerator: (...args: RestArgs<typeof _createGenerator>) => _createGenerator(defaults, ...args),
    resolveConfig: (...args: RestArgs<typeof _resolveConfig>) => _resolveConfig(defaults, ...args),
  }
}
