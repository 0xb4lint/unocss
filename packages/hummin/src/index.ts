export * from '@hummin/core'
export * from '@hummin/preset-default'
export * from '@hummin/preset-attributify'
