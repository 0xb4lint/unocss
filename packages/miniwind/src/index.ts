export * from '@miniwind/core'
export * from '@miniwind/preset-default'
export * from '@miniwind/preset-attributify'
