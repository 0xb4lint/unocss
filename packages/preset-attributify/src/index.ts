import { Preset } from '@miniwind/core'
import { extractorAttributify } from './extractor'
import { AttributifyOptions } from './types'
import { variantAttributify } from './variant'

export * from './extractor'
export * from './variant'
export * from './types'

export const presetAttributify = (options?: AttributifyOptions): Preset => ({
  variants: [
    variantAttributify(options),
  ],
  extractors: [
    extractorAttributify,
  ],
})
