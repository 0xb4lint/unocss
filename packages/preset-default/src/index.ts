import { createUnocssWithDefaults, extractorSplit, Preset } from '@unocss/core'
import { defaultRules } from './rules'
import { defaultTheme } from './theme'
import { defaultVariants } from './variants'

export { defaultTheme, defaultColors } from './theme'

export const presetDefault = (): Preset => ({
  rules: defaultRules,
  variants: defaultVariants,
  extractors: [extractorSplit],
})

export const {
  createGenerator,
  resolveConfig,
} = createUnocssWithDefaults({
  theme: defaultTheme,
  presets: [
    presetDefault(),
  ],
})
