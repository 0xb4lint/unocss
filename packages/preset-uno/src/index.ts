import { extractorSplit, Preset } from '@unocss/core'
import { rules } from './rules'
import { variants } from './variants'

export const preset = (): Preset => ({
  rules,
  variants,
  extractors: [extractorSplit],
})

export default preset
