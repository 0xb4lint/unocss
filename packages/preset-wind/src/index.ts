import { Preset } from '@unocss/core'

export const preset = (): Preset => ({
  // TODO:
})

export default preset
