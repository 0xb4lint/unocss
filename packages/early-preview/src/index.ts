import { createHash } from 'crypto'
import { UserConfig } from '@miniwind/core'
import { presetAttributify } from '@miniwind/preset-attributify'
import { presetDefault } from '@miniwind/preset-default'
import plugin from '@miniwind/vite'

function getHash(input: string, length = 8) {
  return createHash('sha256')
    .update(input)
    .digest('hex')
    .substr(0, length)
}

export default (obj: UserConfig, key: string) => {
  // 0dc2327d
  if (key !== getHash('mwmwmw'))
    return null

  return plugin({
    presets: [
      presetAttributify(),
      presetDefault(),
    ],
    ...obj,
  })
}
