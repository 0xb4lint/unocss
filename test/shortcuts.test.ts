import { createGenerator } from '../packages/preset-default/src'

describe('shortcuts', () => {
  const generate = createGenerator({
    shortcuts: [
      ['sh1', 'px-2 py-3'],
      ['sh2', 'hover:text-lg text-sm text-lg'],
      ['sh3', 'sm:m-2 m-3'],
    ],
  })

  test('generate', async() => {
    const { css } = await generate('sh1 sh2 focus:sh2 sh3')
    expect(css).toMatchSnapshot()
  })
})
