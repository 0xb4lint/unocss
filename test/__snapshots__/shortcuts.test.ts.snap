// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shortcuts generate 1`] = `
".sh1{padding-left:0.5rem;padding-right:0.5rem;padding-top:0.75rem;padding-bottom:0.75rem;}
.sh3{margin:0.75rem;}
.sh2:hover, .focus\\\\:sh2:hover:focus{font-size:1.125rem;line-height:1.75rem;}
.sh2, .focus\\\\:sh2:focus{font-size:0.875rem;line-height:1.25rem;font-size:1.125rem;line-height:1.75rem;}
@media (min-width: 640px){
.sh3{margin:0.5rem;}
}"
`;
